test/SILOptimizer/string_switch.sil

```diff
@@ -0,0 +1,354 @@
+// RUN: %target-sil-opt -enable-sil-verify-all -sil-combine %s | %FileCheck %s
```

Comment:
Hm, how can this test work without a "-string-switch" in the command line here?

It's good that you added a test for out-of-order blocks.
But such a test with _a lot_ of SIL is problematic, because it's basically unmaintainable.
Is it really necessary to include the body of all the functions (beside @test) to make the test work?

Reply:
¯\_(ツ)_/¯ My guess is that `-sil-combine` runs a whole group of passes. When I pass it `-string-switch` it errors `Unknown command line argument` (that is what it's called in the pass def, though).

It seems that `_findStringSwitchCase` isn't in the module unless there is also a string enum. I can remove most of this code if there is some way to fix that (which I am sure there is).